### DISCUSSION TITLE
How to add use client to inline function

### DISCUSSION THREAD

**Discussion by meronogbai** [vercel/next.js#65218] [Help] [answered]
### Summary

Is there a way to mark `MyApp` as a client component here?

```tsx
export default async Page() {
  const MyApp = someComputation();
  return <div><h1>hello</h1><MyApp /></div>
}
```

**Comment by icyJoseph** [comment]
Where does `someComputation` come from?

**Reply by meronogbai** [reply]
@icyJoseph it's custom logic that generates a function 👀. It's kinda complex so can't get into that here.

**Reply by icyJoseph** [reply]
yeah, but if that would live on its own module, then that module could have the `'use client';`

**Reply by meronogbai** [reply]
So I tried that but I'm getting `someComputation` is not a function errors. Give me a few mins, I'll make an example repo.

**Reply by meronogbai** [reply]
Yo sorry for wasting your time. I fixed the issue by moving where where someComputation gets called to a client component. Thanks for the direction ❤️

**Reply by meronogbai** [reply, marked as answer]
Here's the commit if you're interested:

https://github.com/meronogbai/reproduce-error-some-computation/commit/dc19579354e1570a759d089afa193c77f752189a
